### PR TITLE
Merging Clarity<->MARF with Top Level Expression Sorting

### PR DESCRIPTION
**Pull request by kantai**
This is a PR for handling the conflicts between #1084 and #1079

### CODE REVIEW

**Comment by lgalabru** [comment]
Thank you so much for handling this @kantai!
So, one thing that I noticed, and that was missing from https://github.com/blockstack/blockstack-core/pull/1074, is that in this function:
https://github.com/blockstack/blockstack-core/blob/ac07bc24bbe0ef6daf68de801fc577667a14b145/src/vm/checker/typecheck/contexts.rs#L300-L328

Now available here:
https://github.com/blockstack/blockstack-core/blob/1f3d84a00ad7b33b910a0804843fd96ec71ec7a7/src/vm/analysis/check_typing/contexts.rs#L156-L181

We should also grab the fungible and non fungible tokens, and augment the `ContractAnalysis`
Beside that, looks good to me!

**Comment by kantai** [comment]
Addressed in 9daee577 -- I also made this operation destructive (eliminating clones), since the CheckTyping object can be destroyed after the pass (now that we have the nicer `run_pass` functions!)

File: src/vm/database/marf.rs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        .or_else(|| marf.chain_tips().get(0).cloned())
+        .unwrap_or_else(|| TrieFileStorage::block_sentinel());
+
+
+    Ok( MarfedKV { chain_tip, marf, side_store } )
+}
+
+impl MarfedKV {
+    pub fn begin(&mut self, current: &BlockHeaderHash, next: &BlockHeaderHash) {
+        self.marf.begin(current, next)
```
Comment (jcnelson) [inline src/vm/database/marf.rs]: Should we handle an error here more gracefully than just crashing?  Can we at least have an explicit `panic!()`?

Comment (jcnelson) [reply src/vm/database/marf.rs]: Same goes for all these methods -- if we're going to crash, then let's at least give an intelligent reason.

Comment (kantai) [reply src/vm/database/marf.rs]: Sure -- will change these to `.expect`.

**Pull request merged**